_spinnaker/pacrd.md

```diff
@@ -0,0 +1,372 @@
+---
+layout: post
+title: PaCRD
+order: 171
+---
+
+<!-- Note:
+  Setting Order as 171 so it shows up at the bottom on the Spinnaker section
+  of the Armory docs, given it's experimental status. When/if this product
+  hits GA we probably want to move it further up the list and/or replace
+  Dinghy docs in time.
+!-->
+
+PaCRD is considered [**Experimental**] software. This feature is working and
+installable but is missing functionality and subject to rapid change.
+
+__Interested in joining this experiment?
+[Contact us](mailto:akshay.dayal@armory.io) for more information.__
+
+{:.no_toc}
+* This is a placeholder for an unordered list that will be replaced with ToC. To exclude a header, add {:.no_toc} after it.
+{:toc}
+
+# Overview
+
+PaCRD (a portmanteau of "Pipelines as Code" and "Custom Resource Definition") is
+a [Kubernetes controller] that manages the lifecycle of Spinnaker Applications
+and Pipelines as objects within your cluster. It does so by extending Kubernetes
+to support Spinnaker Application and Pipeline objects that can be observed for
+changes via a mature lifecycle management API.
+
+With PaCRD you can:
+
+1. Maintain your Spinnaker Pipelines as code with the rest of your Kubernetes
+manifests.
+1. Persist Pipeline and Application changes with confidence to your Spinnaker
+cluster.
+1. Leverage existing tools like Helm and Kustomize to template your pipelines
+across teams and projects.
+
+
+To get started right away check out the "Quick Start" section for installation
+instructions.
+
+## Pre-Requisites
+
+In order for you to successfully use PaCRD you must first satisfy these
+requirements:
+
+- Have a working Kubernetes 1.11+ cluster
+- Have a working Spinnaker installation
+  - There is no currently identified minimum bound for Spinnaker release
+- Have permissions to install CRDs, create RBAC roles, and create service
+accounts
+
+## Quick Start
+
+This quick start assumes a few things about your Kubernetes cluster:
+
+- Your Spinnaker cluster is installed into a `spinnaker` namespace
+- Your Spinnaker services have `Service` objects of the form `spin-{service_name}`
+  - e.g. `sprint-front50`
+
+Download the current `pacrd` manifest to your local machine:
+
+```
+curl -fsSL https://engineering.armory.io/manifests/pacrd-0.1.0.yaml > pacrd-0.1.0.yaml
+```
+
+Then, inspect the manifest to make sure it's compatible with your cluster.
+When you're ready, apply the `pacrd` manifest to your cluster:
+
+```
+kubectl apply -f pacrd-0.1.0.yaml
```

Comment:
We could tell them to apply to their spinnaker's namespace. Only issue being `ClusterRoleBinding` IIRC

Reply:
or at a minimum inform users of what it is going to do and how to work around it if their spinnaker is in a different namespace than `spinnaker`

Reply:
Showing which lines need to be edited would be sufficient for now

Reply:
Talked about this w/ @ncknt, here's where we landed:
`Sounds like the approach to take is remove namespace from the manifest and define namespace in all the doc commands`

Reply:
Removing the namespace seems like a trickier affair than I'd like to deal with right now. I'm gonna suggest we go with the copy that I've added about removing it from the manifest before applying, then post-launch we can figure out what the best install experience looks like.

Ideally, if/when we open source this we can tell users to run `kustomize` themselves, then they'd have full control over how to install in their cluster!